infrastructure/smart-contracts/test/tinykeys/RefereeBulkSubmissionTransfers.mjs

```diff
@@ -0,0 +1,114 @@
+            expect(winningKeyCountAfterUnstake).to.lt(winningKeyCountAfterAssertion);    
+        });
+
+        it("Check on NodeLicense Transfer, the winning key count changes.", async function () {
+            const { nodeLicense, addr4: owner, referee, challenger, nodeLicenseDefaultAdmin, addr2 } = await loadFixture(deployInfrastructure);
+            
+            const keyBalance = await nodeLicense.balanceOf(owner.address);
+    
+            // Submit a challenge
+            const stateRoot = "0x0000000000000000000000000000000000000000000000000000000000000000";
+            const startingAssertion = 100;
+            await submitTestChallenge(referee, challenger, startingAssertion, stateRoot);
+            const submissionBeforeAssertion = await referee.bulkSubmissions(0, owner.address);
+            const isSubmittedBeforeAssertion = submissionBeforeAssertion[0];
+            const claimedBeforeAssertion = submissionBeforeAssertion[1];
+            const keyCountBeforeAssertion = submissionBeforeAssertion[2];
+            expect(isSubmittedBeforeAssertion).to.equal(false);
+            expect(claimedBeforeAssertion).to.equal(false);
+            expect(keyCountBeforeAssertion).to.equal(0);
+    
+            // Submit Bulk Assertion
+            await referee.connect(owner).submitBulkAssertion(owner.address, 0, stateRoot);
+    
+            // Confirm Bulk Submission Data
+            const submissionAfterAssertion = await referee.bulkSubmissions(0, owner.address);
+            const isSubmittedAfterAssertion = submissionAfterAssertion[0];
+            const claimedAfterAssertion = submissionAfterAssertion[1];
+            const keyCountAfterAssertion = submissionAfterAssertion[2];
+            expect(isSubmittedAfterAssertion).to.equal(true);
+            expect(claimedAfterAssertion).to.equal(false);
+            expect(keyCountAfterAssertion).to.equal(keyBalance);
+
+            const randomBytes = ethers.randomBytes(32);
+            const keysToUnstake = [1n, 2n, 3n];
+
+            const transferRole = await nodeLicense.TRANSFER_ROLE();
+            // Grant the transfer role to the owner address
+            await nodeLicense.connect(nodeLicenseDefaultAdmin).grantRole(transferRole, owner.address);
+            await nodeLicense.connect(owner).adminTransferBatch(addr2, keysToUnstake, randomBytes);
+
+            // Confirm Bulk Submitted Key Count decreased by keys to unstake length    
+            const submissionAfterTransfer = await referee.bulkSubmissions(0, owner.address);
+            const isSubmittedAfterTransfer = submissionAfterTransfer[0];
+            const claimedAfterTransfer = submissionAfterTransfer[1];
+            const keyCountAfterTransfer = submissionAfterTransfer[2];
+            expect(isSubmittedAfterTransfer).to.equal(true);
+            expect(claimedAfterTransfer).to.equal(false);
+            expect(keyCountAfterTransfer).to.equal(keyCountAfterAssertion - BigInt(keysToUnstake.length));    
```

Comment:
Why is this not checking the winningKeyCount ?

Reply:
The goal was to confirm the bulk submission key count was updated on transfer, correct?
If we also want to test the winning key count changing, we'd need to unstake thousands of key ids in this test.

Since the winning key calculation is not in question, the only thing to test here was the key count updating.

Would you prefer to unstake most of the keys and see the winning key count also change? I can quickly add that if preferred.

Reply:
Updated the test to check winning key count, unstake 200 keys and confirm wining key count decreased!